File: back/Application/Services/OrderService.cs
using Application.DTOs;
using Application.Interfaces;
using AutoMapper;
using Domain.Entities;
using Domain.Interfaces;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace Application.Services;

public class OrderService(IOrderDomainService orderDomainService, IMapper mapper) : IOrderService
{
    private readonly IOrderDomainService orderDomainService = orderDomainService;
    private readonly IMapper mapper = mapper;

    public async Task<OrderDTO> GetOrderByIdAsync(int id)
    {
        var order = await orderDomainService.GetOrderByIdAsync(id);
        return mapper.Map<OrderDTO>(order);
    }

    public async Task<int> AddOrderAsync(CreateOrderDTO createOrderDTO)
    {
        var order = mapper.Map<Order>(createOrderDTO);
        return await orderDomainService.AddOrderAsync(order);
    }

    public async Task<PaginatedList<OrderDTO>> GetOrdersAsync(int page, int limit)
    {
        var orders = await orderDomainService.GetPaginatedOrdersAsync(page, limit);
        return new PaginatedList<OrderDTO>
        {
            Items = mapper.Map<List<OrderDTO>>(orders.Items),
            TotalItems = orders.TotalItems,
            TotalPages = orders.TotalPages,
            CurrentPage = orders.CurrentPage
        };
    }
}

File: front/src/Application/services/OrderService.ts
import { OrderServiceInterface, } from "../interfaces";
import { FormModel, OrderDomainServiceInterface } from "@/Domain";

export class OrderService implements OrderServiceInterface {
    private orderDomainService: OrderDomainServiceInterface

    constructor(orderDomainService: OrderDomainServiceInterface) {
        this.orderDomainService = orderDomainService;
    }

    public async getOrders(page: number = 1, limit: number = 10) {
        return await this.orderDomainService.getOrders(page, limit);
    }

    public async getOrderDetail(id: string) {
        return await this.orderDomainService.getOrderDetail(id);
    }

    public async createOrder(data: FormModel) {
        return await this.orderDomainService.createOrder(data);
    }
};